x-pack/auditbeat/processors/sessionmd/processdb/db.go

```diff
@@ -175,22 +179,53 @@ type DB struct {
 	entryLeaderRelationships map[uint32]uint32
 	procfs                   procfs.Reader
 	stopChan                 chan struct{}
-	removalCandidates        rcHeap
-}
-
-func NewDB(reader procfs.Reader, logger logp.Logger) (*DB, error) {
+	// map of processes that we can remove during the next reaper run, if the exit event is older than `removalCandidateTimeout`
+	removalMap map[uint32]removalCandidate
+	ctx        context.Context
+
+	// used for metrics reporting
+	stats *Stats
+
+	// knobs for the reaper thread follows
+
+	// how often the reaper checks for expired or orphaned events.
+	// A negative value disables the reaper.
+	reaperPeriod time.Duration
+	// Tells the reaper to remove orphaned process exec events.
+	// If true, exec events for which no /proc entry can be found will be removed after their insertion time has passed `orphanTimeout`.
+	// If disabled, the reaper will only remove exec events if they are matched with a exit event.
+	reapProcesses bool
+	// The duration after which we'll reap an orphaned process exec event for which no /proc data exists. Measured from the time the event is inserted.
+	processReapAfter time.Duration
+}
+
+// NewDB creates a new DB for tracking processes.
+// metrics: monitoring registry for exporting DB metrics
+// reader: handler for /proc data and events.
+// reaperPeriod: tells the reaper to update its tracking of deat and orphaned processes every at every `n` period.
+// reapProcesses: optionally tell the reaper to also reap orphan processes from the DB, if no matching exit event can be found. May result in data loss if the DB in under load and events do not arrive in a timely fashion.
```

Comment:
This isn't a conventional format for a [function go doc](https://tip.golang.org/doc/comment#func), but adding the indented bullets renders much better and is easier to read in godoc.

```suggestion
// NewDB creates a new DB for tracking processes.
//
//   - metrics: monitoring registry for exporting DB metrics
//   - reader: handler for /proc data and events.
//   - reaperPeriod: tells the reaper to update its tracking of dead and
//     orphaned processes at every `n` period.
//   - reapProcesses: optionally tell the reaper to also reap orphan processes
//     from the DB, if no matching exit event can be found. May result in data
//     loss if the DB in under load and events do not arrive in a timely fashion.
```

It also fixes typo/grammar in the `reaperPeriod` section.